From 85a6cd1f62f6b31fb21ddea9e01da97e8c940e0d Mon Sep 17 00:00:00 2001
From: Martin Pitt <martinpitt@gnome.org>
Date: Fri, 24 May 2013 09:35:44 +0200
Subject: [PATCH 3/3] dosfslabel: Do not read beyond string length

When checking whether the label contains any lower-case characters, do not read
beyond the end of the string.
---
 src/dosfslabel.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/dosfslabel.c b/src/dosfslabel.c
index dd8d36a..c87c834 100644
--- a/src/dosfslabel.c
+++ b/src/dosfslabel.c
@@ -115,7 +115,7 @@ int main(int argc, char *argv[])
 		    "dosfslabel: labels can be no longer than 11 characters\n");
 	    exit(1);
 	}
-        for (i = 0; i < 11; i++)
+        for (i = 0; label[i] && i < 11; i++)
           /* don't know if here should be more strict !uppercase(label[i])*/
           if (islower(label[i])) {
             fprintf(stderr,
-- 
1.8.1.2

